src/main/java/se/kits/gakusei/util/csv/CSV.java

```diff
@@ -0,0 +1,69 @@
+package se.kits.gakusei.util.csv;
+
+import com.univocity.parsers.common.processor.RowListProcessor;
+import com.univocity.parsers.csv.CsvParser;
+import com.univocity.parsers.csv.CsvParserSettings;
+import se.kits.gakusei.util.ParserFailureException;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class CSV {
+
+    /**
+     * Parse a CSV file and return a map with its headers and rows.
+     *
+     * @param filePath
+     * @param expectedNbrOdHeaders
+     * @return Map
+     */
+    public static Map<String, List<String[]>> parse(String filePath, int expectedNbrOdHeaders) {
+
+        CsvParserSettings settings = new CsvParserSettings();
+        RowListProcessor rowProcessor = new RowListProcessor();
+
+        settings.setLineSeparatorDetectionEnabled(true);
+        settings.setRowProcessor(rowProcessor);
+        settings.setHeaderExtractionEnabled(true);
+
+        CsvParser parser = new CsvParser(settings);
+
+        Map<String, List<String[]>> result = new HashMap<String, List<String[]>>();
+
+        try {
+            parser.parse(new FileReader(filePath));
+        } catch(IOException e) {
+
```

Comment:
Maybe throw parserFailureException here instead of embedding error in the return value?

Reply:
good idea! ✅